realtime_connext/poll_utils.cxx

```diff
@@ -0,0 +1,468 @@
+        printf("create_subscriber error\n");
+        subscriber_shutdown(participant);
+        return -1;
+    }
+
+    /* Register the type before creating the topic */
+    type_name = pollTypeSupport::get_type_name();
+    retcode = pollTypeSupport::register_type(
+        participant, type_name);
+    if (retcode != DDS_RETCODE_OK) {
+        printf("register_type error %d\n", retcode);
+        subscriber_shutdown(participant);
+        return -1;
+    }
+
+    /* To customize the topic QoS, use 
+       the configuration file USER_QOS_PROFILES.xml */
+    topic = participant->create_topic(
+        "Example poll",
+        type_name, DDS_TOPIC_QOS_DEFAULT, NULL /* listener */,
+        DDS_STATUS_MASK_NONE);
+    if (topic == NULL) {
+        printf("create_topic error\n");
+        subscriber_shutdown(participant);
+        return -1;
+    }
+
+    /* Call create_datareader passing NULL in the listener parameter */
+    reader = subscriber->create_datareader(
+        topic, DDS_DATAREADER_QOS_DEFAULT, NULL,
+        DDS_STATUS_MASK_ALL);
+    if (reader == NULL) {
+        printf("create_datareader error\n");
+        subscriber_shutdown(participant);
+        return -1;
+    }
+
+    /* If you want to change datareader_qos.history.kind programmatically rather
+     * than using the XML file, you will need to add the following lines to your
+     * code and comment out the create_datareader call above. */
+
+    /*
+    DDS_DataReaderQos datareader_qos;
+    retcode = subscriber->get_default_datareader_qos(datareader_qos);
+    if (retcode != DDS_RETCODE_OK) {
+        printf("get_default_datareader_qos error\n");
+        return -1;
+    }
+
+    datareader_qos.history.kind = DDS_KEEP_ALL_HISTORY_QOS;
+
+    reader = subscriber->create_datareader(
+        topic, datareader_qos, NULL,
+        DDS_STATUS_MASK_ALL);
+    if (reader == NULL) {
+        printf("create_datareader error\n");
+        subscriber_shutdown(participant);
+        return -1;
+    }
+    */
+
+    pollDataReader *poll_reader = pollDataReader::narrow(reader);
+    if (poll_reader == NULL) {
+        printf("DataReader narrow error\n");
+        return -1;
+    }
+
+    if (rttest_set_sched_priority(97, SCHED_RR) != 0) {
+      perror("Failed to set scheduling priority");
+    }
+
+    if (rttest_lock_memory() != 0) {
+      perror("Couldn't lock memory");
+    }
+    rttest_prefault_stack();
+
+    /* Main loop */
+    rttest_spin(subscriber_callback, static_cast<void*>(poll_reader));
+
+    /* Delete all entities */
+    status = subscriber_shutdown(participant);
+
+    rttest_write_results_file("rttest_subscriber_results");
+    rttest_finish();
+
+    return status;
+}
+
+int subscriber_main(Arguments *args)
+{
+    return subscriber_main(args->argc, args->argv);
+}
+
+void *subscriber_main(void *args)
+{
+    Arguments *new_args = static_cast<Arguments*>(args);
+    int ret = subscriber_main(new_args);
+    return static_cast<void*>(&ret);
```

Comment:
Looks like you're returning a pointer to a temporary object (`ret`). What's trying to be accomplished here?